### DISCUSSION TITLE
Feature request: automatic GITHUB_TOKEN in workflows that can trigger events

### DISCUSSION THREAD

**Discussion by stefanfoulis** [community/community#35047] [Actions]
I understand that by default the automatically generated GITHUB_TOKEN in workflows will not trigger any further workflows as a side effect from apis that it calls. As far as I understood the main reason is to avoid accidental recursive calling of workflows.
I think this is a very reasonable default.

The workaround for cases where it actually is desireable that api requests would trigger further workflows is to use a Personal Access Token (PAT). While this works, it is not ideal because it is not possible to limit permissions of a PAT in the same way the automatic GITHUB_TOKEN is limited. And it is cumbersome to setup.

My suggestion is to add a optional [job/workflow permission](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) which would enable triggering further workflows as a side-effect of api calls.

**Comment by peter-evans** [comment]
I also had this same idea and came here to request this feature.

It would be great if we can choose to elevate the permission of the default `GITHUB_TOKEN` to allow it to trigger further workflow runs. 

For example, there could be a new permission called `workflow-events` (or whatever works best in GitHub terminology). This would be `none` by default, and setting it to `write` would allow the token to trigger further workflows.

```yml
permissions:
  workflow-events: write
```

**Reply by anishkny** [reply]
@peter-evans kudos for bending over backwards and providing multiple [workarounds](https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs) for this limitation in GitHub.

Unfortunately they all seem hacky and have some drawback or another. This should really be supported natively by the platform. 🤞

**Comment by mhils** [comment]
If your use case is to automatically update/fix the current branch or pull request, you can use the [autofix.ci action](https://autofix.ci/). 

Of course it would be fantastic if this becomes a GitHub feature and I don't have to maintain it anymore. 😃
